modules/database/src/main/java/com/intuit/wasabi/database/impl/DBITransaction.java

```diff
@@ -56,29 +56,31 @@ public DBITransaction(DBI dbi) {
      */
     @Override
     public Object transaction(Block block) {
+        Handle handle = null;
         try {
-            begin();
+            handle = getHandle();
```

Comment:
All the methods following this call depend on the premise that getHandle() is never going to return a null handle, otherwise, each of the following calls run the risk of throwing a NPE. Possibly a good idea to check for handle being null in getHandle() and wrapping the NPE appropriately.